metricbeat/module/windows/perfmon/reader.go

```diff
@@ -74,9 +75,10 @@ func NewReader(config Config) (*Reader, error) {
 				return nil, errors.Wrapf(err, `failed to expand counter (query="%v")`, counter.Query)
 			}
 		}
-		if childQueries == nil || len(childQueries) == 0 {
+		// check if the pdhexpandcounterpath/pdhexpandwildcardpath functions have expanded the counter successfully.
+		if len(childQueries) == 0 || (len(childQueries) == 1 && strings.Contains(childQueries[0], "*")) {
 			query.Close()
-			return nil, errors.Wrapf(err, `failed to expand counter (query="%v")`, counter.Query)
```

Comment:
Good catch here btw :slightly_smiling_face: I guess it was failing silently before? (`errors.Wrap(nil)` returns `nil`)

Reply:
@jsoriano , yes, found it while running the tests.